File: NeuronNetworks/Perseptron/Layers/InputLayer.cs
using NeuronNetworks.Perseptron.Interfaces;

namespace NeuronNetworks.Perseptron.Layers
{
    class InputLayer : IInputLayer
    {
        public InputLayer(int inputsCount, int index)
        {
            InputsCount = inputsCount;
            Index = index;
        }

        public int Index { get; }
        public ILayer NextLayer { get; set; }
        public int InputsCount { get; }

        public int OutputsCount => InputsCount;

        public InputOutputModel GetInputAndOutput(double[] inputs)
        {
            return new InputOutputModel(inputs, inputs);
        }

        public double[] Handle(double[] inputs)
        {
           return NextLayer.Handle(inputs);
        }
    }
}

File: JavaScript (TypeScript)/src/neuron-network/perseptron/Layers/input-layer.ts
import { ILayer } from '../../abstract';
import { IInputLayer } from '../abstract';

export class InputLayer implements IInputLayer {
    nextLayer: ILayer;
    outputsCount: number;

    /**
     * Input layer
     */
    constructor(public readonly inputsCount: number, public readonly index: number) {
        this.outputsCount = inputsCount;
    }

    handle(inputs: number[]): number[] {
        return inputs;
    }
}
